src/lib/protocols/quic.c

```diff
@@ -31,107 +31,135 @@
 #define QUIC_SEQ_MASK 0x30      // 0011 0000
 
 #define CID_LEN_8 0x0C          // 0000 1100
+#define CID_LEN_0 0x00          // 0000 0000
+
 #define CID_LEN_4 0x08          // 0000 1000
 #define CID_LEN_1 0x04          // 0000 0100
-#define CID_LEN_0 0x00          // 0000 0000
 
 #define SEQ_LEN_6 0x30          // 0011 0000
 #define SEQ_LEN_4 0x20          // 0010 0000
 #define SEQ_LEN_2 0x10          // 0001 0000
 #define SEQ_LEN_1 0x00          // 0000 0000
 
-#define SEQ_CONV(ARR) (ARR[0] | ARR[1] | ARR[2] | ARR[3] | ARR[4] | ARR[5] << 8)
-
+#define INT(C) (C - '0')
+#define DIGIT(X, Y, Z) ((isdigit(X) && isdigit(Y) && isdigit(Z)) ? (INT(X) * 100 + INT(Y) * 10 + INT(Z)) : 0)
 
 #ifdef NDPI_PROTOCOL_QUIC
-static void ndpi_int_quic_add_connection(struct ndpi_detection_module_struct
-                                         *ndpi_struct, struct ndpi_flow_struct *flow)
+static void ndpi_int_quic_add_connection(struct ndpi_detection_module_struct *ndpi_struct, struct ndpi_flow_struct *flow)
 {
-  ndpi_set_detected_protocol(ndpi_struct, flow, NDPI_PROTOCOL_QUIC, NDPI_PROTOCOL_UNKNOWN);
+    ndpi_set_detected_protocol(ndpi_struct, flow, NDPI_PROTOCOL_QUIC, NDPI_PROTOCOL_UNKNOWN);
 }
 
 static int connect_id(const unsigned char pflags)
 {
     u_int cid_len;
 
-        // Check CID length.
-        switch (pflags & QUIC_CID_MASK)
-        {
-           case CID_LEN_8: cid_len = 8; break;
-           case CID_LEN_4: cid_len = 4; break;
-           case CID_LEN_1: cid_len = 1; break;
-           case CID_LEN_0: cid_len = 0; break;
-           default:
-               return -1;
+    // Check CID length
+    switch (pflags & QUIC_CID_MASK)
+    {
+    	case CID_LEN_8: cid_len = 8; break;
```

Comment:
We are discussing about to completely remove this two-bytes control or add a support for older versions. For now, in **Q032** still exist.